OCR.py

```diff
@@ -0,0 +1,162 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.image import imread
+from PIL import Image
+import tempfile
+import math
+import logging
+import string
+char_whitelist = string.digits
+char_whitelist += string.ascii_lowercase
+char_whitelist += string.ascii_uppercase
+char_whitelist += string.punctuation.replace("'","").replace('"','')
+
+
+def load_image( infilename ) :
+    """loads an image file to an array
+    Args:
+        infilename: file path
+    Returns:
+        numpy array of image data
+    """
+    im = imread(infilename)
+    return np.array(im)
+
+def save_image( npdata, outfilename ) :
+    """saves an image file from an array
+    Args:
+        npdata: (array) numpy array of image data
+        outfilename: (str)strfile path
+    Returns:
+        None
+    """
+    outimg = Image.fromarray( npdata, "RGB" )
+    outimg.save( outfilename )
+
+
+def cut_metadata(im, camera_type):
+    """locates and cuts the metadata tag from image
+    Args:
+        im (array) : numpy array of image data
+        camera_type (str): {'hpwren','Axis','unknown'} defined type of image to remove metadata from.
+    Returns:
+        metadatastrip (array): numpy array containing only the presumed metadata line
+    """
+    #uses first order central difference gradient in 1-D to determine edges of text
+    if camera_type == 'unknown':
+        index = im.shape[1]//2
+        if np.sum(im[im.shape[0]-1,im.shape[1]//2,:]) < 80:#check if metadata bar is present on bottom
+            camera_type = 'Axis'
+        else:
+            camera_type = 'hpwren'
+
+    if camera_type == 'Axis':
```

Comment:
How different are these values across the Axis cameras and have you measured how much extra compute this calculation takes up?  It may be worth caching the results.